### PR TITLE
feat: add pixi global list filter

### PR DESCRIPTION
**Pull request by nichmor**
Add optional `regex` argument for both` pixi global list` and `pixi global list --environment env-name`.

If passed regex string is not a regex syntax, but a simple string - it will act as a contains filter.

Some examples:

Having `nushell`, `python`, `python3-10` as enviroments

`pixi global list py`

Screenshot 2024-10-14 at 16 49 11

`pixi global list --environment python lib`

Screenshot 2024-10-14 at 16 52 04

### CODE REVIEW

File: tests/integration/test_global.py [hunk 1]
```diff
@@ -863,6 +863,33 @@ def test_list(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
...
+        ],
+        env=env,
+    )
+
+    # Verify list with dummy-b
+    verify_cli_command(
+        [pixi, "global", "list", "dummy-a"],
+        env=env,
+        stdout_contains=["dummy-a: 0.1.0", "dummy-a", "dummy-aa"],
+    )
```
Comment (Hofer-Julian) [inline tests/integration/test_global.py:890]: Could you add a test for `pixi global list --environment dummy-a dummy` and make sure that it only contains `dummy-a` and not `dummy-b`?

**Review by Hofer-Julian — COMMENTED** [review]
Works like a charm!

I had a few comments

File: tests/integration/test_global.py (hunk 1 above)
Comment (nichmor) [reply tests/integration/test_global.py:890]: yeah make sense - let me do this

File: tests/integration/test_global.py [hunk 2]
```diff
@@ -863,6 +863,33 @@ def test_list(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
...
+            "dummy-a==0.1.0",
+        ],
+        env=env,
+    )
+
+    # Verify list with dummy-b
+    verify_cli_command(
+        [pixi, "global", "list", "dummy-a"],
+        env=env,
+        stdout_contains=["dummy-a: 0.1.0", "dummy-a", "dummy-aa"],
```
[the remark this answers is not in the document]
Comment (nichmor) [reply tests/integration/test_global.py:889]: sure - good idea

**Review by Hofer-Julian — APPROVED** [review]
One suggestion, then we can merge this!

File: tests/integration/test_global.py [hunk 3]
```diff
@@ -863,6 +863,43 @@ def test_list(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
...
+        stdout_excludes=["dummy-b"],
+    )
+
+    # Verify list filter for environment dummy-a.
+    # It should not contains dummy-b
+    verify_cli_command(
+        [pixi, "global", "list", "--environment", "dummy-a", "dummy-a"],
```
[the remark this answers is not in the document]
Comment (nichmor) [reply tests/integration/test_global.py:896]: well thought!

**Pull request merged**